### PR TITLE
Docker for local dev

### PR DESCRIPTION
**Pull request by shannonrdunn**

### CODE REVIEW

**Comment by lelandg559** [comment]
Not sure if this is still expected to come from the docker container api server 

could not connect to server: No such file or directory Is the server running locally and accepting connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: '2'
+services:
+  db:
+    image: postgres
+  web:
+    build: .
+    ports:
+      - "3000:3000"
```
Comment (lelandg559) [inline docker-compose.yml]: this maybe should be port 5432?

**Comment by shannonrdunn** [comment]
@lelandg559 can you try these latest changes?

**Comment by shannonrdunn** [comment]
not sure why the travis build is failing, but give it a shot ill fix the travis error

**Comment by lelandg559** [comment]
worked for me, let me look into 

$ exec rake rubocop db:schema:load test
rake aborted!
Gem::LoadError: You have already activated rake 10.4.2, but your Gemfile requires rake 11.1.0. Prepending `bundle exec` to your command may solve this. 

before we merge,

**Comment by jfhamlin** [comment]
Yeah, looks like the branch needs to be rebased onto master.

**Comment by shannonrdunn** [comment]
I think thats how you rebase ;)

**Comment by shannonrdunn** [comment]
Just need to fix the db hostname problem on travis

File: .travis.yml [hunk 2]
```diff
@@ -12,6 +12,8 @@ addons:
   # this is working out-of-the-box on Travis, so we'll go with it in
   # CI for now.
   postgresql: "9.2"
+env:
+- RAILS_ENV=test
```
Comment (jfhamlin) [inline .travis.yml]: Is this addition needed?

Comment (shannonrdunn) [inline .travis.yml]: https://github.com/ShelterTechSF/askdarcel-api/blob/master/config/database.yml#L58

Trying to get travis to run in this DB context. And not use the db hostname we set up for the development environment. (why the travis build is failing), looks like it is still picking up the hostname that is really only useful in docker.

**Comment by lelandg559** [comment]
getting this on the latest, not sure if its just me

leland@ophion:~/askdarcel-api$ docker-compose up
Starting askdarcelapi_db_1
Starting askdarcelapi_web_1
Attaching to askdarcelapi_db_1, askdarcelapi_web_1
db_1  | LOG:  database system was shut down at 2016-03-24 02:26:28 UTC
db_1  | LOG:  MultiXact member wraparound protections are now enabled
db_1  | LOG:  database system is ready to accept connections
db_1  | LOG:  autovacuum launcher started
web_1 | /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/spec_set.rb:94:in `block in materialize': Could not find diff-lcs-1.2.5 in any of the sources (Bundler::GemNotFound)
web_1 | 	from /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/spec_set.rb:87:in `map!'
web_1 | 	from /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/spec_set.rb:87:in `materialize'
web_1 | 	from /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/definition.rb:137:in `specs'
web_1 | 	from /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/definition.rb:182:in `specs_for'
web_1 | 	from /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/definition.rb:171:in `requested_specs'
web_1 | 	from /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/environment.rb:18:in `requested_specs'
web_1 | 	from /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/runtime.rb:13:in `setup'
web_1 | 	from /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler.rb:92:in `setup'
web_1 | 	from /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.11.2/lib/bundler/setup.rb:18:in `<top (required)>'
web_1 | 	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:55:in `require'
web_1 | 	from /usr/local/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:55:in `require'
askdarcelapi_web_1 exited with code 1

File: .travis.yml [hunk 3]
```diff
@@ -13,6 +13,6 @@ addons:
   # CI for now.
   postgresql: "9.2"
 
-before_script: bundle exec rake db:create
+before_script: RAILS_ENV=test bundle exec rake db:create
```
Comment (jfhamlin) [inline .travis.yml]: I think this environment variable is set by default by Travis: https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables
